client/state/plugins/installed/actions.js

```diff
@@ -0,0 +1,360 @@
+/**
+ * External dependencies
+ */
+import wpcom from 'lib/wp';
+
+/**
+ * Internal dependencies
+ */
+import utils from 'lib/site/utils';
+import {
+	PLUGINS_RECEIVE,
+	PLUGINS_REQUEST,
+	PLUGINS_REQUEST_SUCCESS,
+	PLUGINS_REQUEST_FAILURE,
+	PLUGIN_ACTIVATE_REQUEST,
+	PLUGIN_ACTIVATE_REQUEST_SUCCESS,
+	PLUGIN_ACTIVATE_REQUEST_FAILURE,
+	PLUGIN_DEACTIVATE_REQUEST,
+	PLUGIN_DEACTIVATE_REQUEST_SUCCESS,
+	PLUGIN_DEACTIVATE_REQUEST_FAILURE,
+	PLUGIN_UPDATE_REQUEST,
+	PLUGIN_UPDATE_REQUEST_SUCCESS,
+	PLUGIN_UPDATE_REQUEST_FAILURE,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST_SUCCESS,
+	PLUGIN_AUTOUPDATE_ENABLE_REQUEST_FAILURE,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST_SUCCESS,
+	PLUGIN_AUTOUPDATE_DISABLE_REQUEST_FAILURE,
+	PLUGIN_INSTALL_REQUEST,
+	PLUGIN_INSTALL_REQUEST_SUCCESS,
+	PLUGIN_INSTALL_REQUEST_FAILURE,
+	PLUGIN_REMOVE_REQUEST,
+	PLUGIN_REMOVE_REQUEST_SUCCESS,
+	PLUGIN_REMOVE_REQUEST_FAILURE
+} from 'state/action-types';
+import {
+	ACTIVATE_PLUGIN,
+	DEACTIVATE_PLUGIN,
+	UPDATE_PLUGIN,
+	ENABLE_AUTOUPDATE_PLUGIN,
+	DISABLE_AUTOUPDATE_PLUGIN,
+	INSTALL_PLUGIN,
+	REMOVE_PLUGIN
+} from './constants';
+
+/**
+ * Return plugin id depending if the site is a jetpack site
+ *
+ * @param {Object} site - site object
+ * @param {Object} plugin - plugin object
+ * @return {String} plugin if
+ */
+const getPluginId = ( site, plugin ) => {
+	return site.jetpack ? plugin.id : plugin.slug;
+};
+
+/**
+ * Return a SitePlugin instance used to handle the plugin
+ *
+ * @param {Object} site - site object
+ * @param {String} pluginId - plugin identifier
+ * @return {SitePlugin} SitePlugin instance
+ */
+const getPluginHandler = ( site, pluginId ) => {
+	const siteHandler = wpcom.site( site.ID );
+	const pluginHandler = site.jetpack
+		? siteHandler.plugin( pluginId )
+		: siteHandler.wpcomPlugin( pluginId );
+
+	return pluginHandler;
+};
+
+export function activate( site, plugin ) {
```

Comment:
I think the naming of the action creators should be more specific to the context to avoid namespace conflicts if either several action creators are imported within the same file (without needing to rename them during import) and if we were ever to do something like [global selectors](https://github.com/Automattic/wp-calypso/pull/5954) for action creators.